component/src/main/java/org/wso2/extension/siddhi/execution/env/ResourceBatchWindowProcessor.java

```diff
@@ -0,0 +1,383 @@
+                                attributeExpressionExecutors[2]).getValue()));
+                    } else {
+                        throw new SiddhiAppValidationException("ResourceBatch window's 3rd parameter " +
+                                "'time.in.milliseconds' should be either be a constant (of type int or long), " +
+                                "but found " + attributeExpressionExecutors[2].getReturnType());
+                    }
+                } else {
+                    throw new SiddhiAppValidationException("ResourceBatch window's 3rd parameter " +
+                            "'time.in.milliseconds' should either be a constant (of type int or long)");
+                }
+            }
+        } else {
+            throw new SiddhiAppValidationException(
+                    "Resource batch window should only have two or three parameters, but found "
+                            + attributeExpressionExecutors.length + " input attributes");
+        }
+    }
+
+    @Override
+    protected void process(ComplexEventChunk<StreamEvent> streamEventChunk, Processor nextProcessor,
+                           StreamEventCloner streamEventCloner) {
+        synchronized (this) {
+            long currentTime = siddhiAppContext.getTimestampGenerator().currentTime();
+            ComplexEventChunk<StreamEvent> outputStreamEventChunk = new ComplexEventChunk<StreamEvent>(true);
+            while (streamEventChunk.hasNext()) {
+                StreamEvent streamEvent = streamEventChunk.next();
+                StreamEvent clonedStreamEvent = streamEventCloner.copyStreamEvent(streamEvent);
+                if (!streamEvent.getType().equals(ComplexEvent.Type.TIMER)) {
+                    Object groupEventMapKey = groupKeyExpressionExecutor.execute(clonedStreamEvent);
+                    ResourceStreamEventList resourceStreamEventList = groupEventMap.get(groupEventMapKey);
+                    if (resourceStreamEventList != null) {
+                        resourceStreamEventList.streamEventList.add(clonedStreamEvent);
+                    } else {
+                        List<StreamEvent> list = new ArrayList<>();
+                        list.add(clonedStreamEvent);
+                        groupEventMap.put(groupEventMapKey, new ResourceStreamEventList(list,
+                                clonedStreamEvent.getTimestamp() + timeInMilliSeconds));
+                        long newNextEmitTime = currentTime + timeInMilliSeconds;
+                        if (nextEmitTime == -1 || newNextEmitTime > nextEmitTime) {
+                            nextEmitTime = newNextEmitTime;
+                            if (scheduler != null) {
+                                scheduler.notifyAt(nextEmitTime);
+                            }
+                        }
+                    }
+                }
+                if (outputExpectsExpiredEvents) {
```

Comment:
remove if condition and have the if part only

Reply:
Fixes added: d6242876caf0f65c7fd023c25c7c5e50e9fd7b3c